src/components/Toolbar.tsx

```diff
@@ -0,0 +1,388 @@
+          track.enabled = true;
+        });
+        // TODO: setLocalVideoText(disableLocalVidDrag ? '' : 'Drag Me');
+        setLocalVideoText('');
+      }
+      setLocalStream(localStream);
+    }
+  };
+
+  const handleSharing = (
+    sharing: boolean,
+    setSharing: Function,
+    connected: Map<string, boolean>,
+    peerConnections: Map<string, RTCPeerConnection>,
+    switchInputDevices: Function
+  ) => {
+    if (!isConnected(connected)) {
+      logger('You must join a call before you can screen share');
+      return;
+    }
+    if (!sharing) {
+      const gdmOptions = {
+        video: true,
+        audio: {
+          echoCancellation: true,
+          noiseSuppression: true,
+          sampleRate: 44100,
+        },
+      };
+      navigator.mediaDevices
+        .getDisplayMedia(gdmOptions)
+        .then((stream: MediaStream) => {
+          setSharing(true);
+          if (stream.getAudioTracks()[0])
+            stream.addTrack(stream.getAudioTracks()[0]);
+          logger(stream.toString());
+          // swap audio/video streams
+          let videoTrack = stream.getVideoTracks()[0];
+          let audioTrack = stream.getAudioTracks()[0];
+          connected.forEach(
+            (value: boolean, uuid: string, map: Map<string, boolean>) => {
+              if (connected.get(uuid)) {
+                const sender = peerConnections
+                  ?.get(uuid)
+                  ?.getSenders()
+                  .find((s: any) => {
+                    return s.track.kind === videoTrack.kind;
+                  });
+                if (sender) sender.replaceTrack(videoTrack);
+                if (stream.getAudioTracks()[0]) {
+                  logger('Audio track is' + audioTrack.toString());
+                  const sender2 = peerConnections
+                    ?.get(uuid)
+                    ?.getSenders()
+                    .find((s: any) => {
+                      if (s.track.kind === audioTrack.kind) {
+                        logger('Found matching track: ' + s.track.toString());
+                      }
+                      return s.track.kind === audioTrack.kind;
+                    });
+                  // add track instead of replacing
+                  if (sender2) sender2.replaceTrack(audioTrack);
+                }
+              }
+            }
+          );
+          setLocalStream(stream);
+        })
+        .catch(err => {
+          logger(err);
+        });
+    } else if (sharing) {
+      setSharing(false);
+      switchInputDevices();
+    }
+  };
+
+  return (
+    <div
+      id="toolbar"
+      ref={toolbarRef}
+      className="fixed min-w-screen max-w-screen bottom-0 left-0 right-0 sm:bottom-2 z-6 flex justify-center flex-row"
+    >
+      <div
+        id="multi-btn"
+        className="sm:rounded-2xl text-2xl bg-white dark:bg-gray-800 shadow-sm px-6 py-3 flex flex-row items-center content-evenly sm:mb-1 min-w-screen max-w-screen w-screen sm:w-auto" // fixed
+      >
```

Comment:
potential improvement in the future that one of us could take on - general toolbar button component